crates/forge/tests/e2e/docs_snippets_validation.rs

```diff
@@ -0,0 +1,43 @@
+use clap::Parser;
+use docs::validation::{extract_matches_from_directory, get_parent_dir, snippet_to_command_args};
+use forge::Cli;
+use regex::Regex;
+#[test]
+fn test_docs_snippets() {
+    let root_dir = get_parent_dir(2);
+    let docs_dir = root_dir.join("docs/src");
+
+    let re = Regex::new(r"(?ms)```shell\n\$ (snforge .+?)\n```").expect("Invalid regex pattern");
+    let extension = Some("md");
+
+    let snippets = extract_matches_from_directory(&docs_dir, &re, extension)
+        .expect("Failed to extract snforge command snippets");
+
+    let skipped_args = [
+        // for some reason `try_parse_from` fails on `--version` flag
+        vec!["snforge", "--version"],
+    ];
+
+    for snippet in &snippets {
+        let args = snippet_to_command_args(snippet.as_str());
+        let args: Vec<&str> = args.iter().map(String::as_str).collect();
+
+        if skipped_args.contains(&args) {
+            continue;
+        }
```

Comment:
I think this should maybe print a warning or something when skipping. I see that eventually someone might extend `skipped_args` with something for whatever reason and we won't be aware of that.